code-samples/eventing/bookstore-sample-app/frontend/client/components/Toggle.js

```diff
@@ -0,0 +1,19 @@
+'use client';
+import React from 'react';
+import { useTheme } from 'next-themes';
+
+const Toggle = () => {
+	const { systemTheme, theme, setTheme } = useTheme();
+	const currentTheme = theme === 'system' ? systemTheme : theme;
+
```

Comment:
```suggestion
  useEffect(() => {
    const systemPreference = window.matchMedia("(prefers-color-scheme: dark)")
      .matches
      ? "dark"
      : "light";

    setTheme(systemPreference);
  }, []);
```
Adding these will allow the bookstore to automatically detect what mode the user is using on their laptop, and use that mode as default.